codegen/src/main/java/software/amazon/awssdk/codegen/model/config/customization/CustomizationConfig.java

```diff
@@ -227,6 +227,18 @@ public class CustomizationConfig {
      */
     private String asyncClientDecorator;
 
+    /**
+     * For S3AsyncClient only. Fully qualified name of the class to be used for multipart configuration
+     */
+    private String multipartConfigurationClass;
+
+    /**
+     * For S3AsyncClient only. Javadoc associated with the generated method on the S3AsyncClientBuilder interface for multipart
+     * configuration.
+     */
+    private String multipartMethodJavadoc;
```

Comment:
What are the pros/cons of this compared to just using customClientContextParams or something similar? https://github.com/aws/aws-sdk-java-v2/blob/master/services/s3/src/main/resources/codegen-resources/customization.config#L255-L260

Reply:
Custom context param also adds the method to the sync client builder, but we only want multipart for the async client

Reply:
Seems fine, since this is just an internal API. We can always refactor it later if we wanted to. E.g. make customClientContextParams have an async-only flag.